### ISSUE TITLE
How to install REMP on WO

### ISSUE THREAD

**Issue by PayteR** [remp2020/remp#8]

**Comment by rootpd** [comment]
Hi,

it's been a while since we tested the whole appliance on Windows and makefiles are surely one of the things that are missing there :/.

First the explanation of what happens. The `make docker-build` prebuilds Go binaries, which are then included within `docker-compose up` to containers. As they weren't built, your `docker-compose up` failed on missing file error.

Now how you can solve it:

* Try to run `make docker-build` on WSL (in bash of windows subsystem for linux). That `make` command should be fully compatible with the definition required to be run. (recommended)

* Run the commands manually (not recommended, as these might change in the future):

```
cd /path/to/remp/Beam/go/cmd/segments
docker build -t remp-beam_segments_builder builder
docker run --rm -v /path/to/remp/Beam/go/:/src/build remp-beam_segments_builder > segments.tar;

cd /path/to/remp/Beam/go/cmd/tracker
docker build -t remp-beam_tracker_builder builder
docker run --rm -v /path/to/remp/Beam/go:/src/build remp-beam_tracker_builder > tracker.tar;
```

As a result, you should see these two file appear on your filesystem:

* `/path/to/remp/Beam/go/cmd/segments/segments.tar`
* `/path/to/remp/Beam/go/cmd/tracker/tracker.tar`

They include binary that's passed to Docker container when `docker-compose up` is called.

**Comment by PayteR** [comment]
Thank you, i have tried that, it `make` command run well, but now i have another issue, where I'm stuck. When i ran `make` command in WSL and then `docker-compose up` in windows command line, then it stuck in cycle with this errors. That are compiled *.go packages that i try to run on windows, so i need to run in on WSL's docker setup?
```
beam_tracker_1   | 2019/06/17 16:39:40 [EROR] unable to cache properties err=Error 1146: Table 'beam.properties' doesn't exist
beam_tracker_1   | unable to cache properties from MySQL
beam_tracker_1   | gitlab.com/remp/remp/Beam/go/model.(*PropertyDB).Cache
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/model/property.go:67
beam_tracker_1   | main.main.func1
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/tracker/main.go:97
beam_tracker_1   | main.main.func3
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/tracker/main.go:118
beam_tracker_1   | runtime.goexit
beam_tracker_1   |      /usr/local/go/src/runtime/asm_amd64.s:1333
beam_segments_1  | 2019/06/17 16:39:44 [EROR] unable to cache segments err=Error 1146: Table 'beam.segments' doesn't exist
beam_segments_1  | unable to cache segments from MySQL
beam_segments_1  | gitlab.com/remp/remp/Beam/go/model.(*SegmentDB).Cache
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/model/segment.go:553
beam_segments_1  | main.main.func1
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/segments/main.go:112
beam_segments_1  | main.main.func5
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/segments/main.go:143
beam_segments_1  | runtime.goexit
beam_segments_1  |      /usr/local/go/src/runtime/asm_amd64.s:1333
beam_segments_1  | ELASTIC 2019/06/17 16:39:44 elastic: http://elasticsearch:9200 is dead
sso_1            |   - Installing symfony/stopwatch (v4.2.4): Downloading (100%)
campaign_1       |   - Installing symfony/options-resolver (v4.2.4): Downloading (100%)
sso_1            |   - Installing symfony/polyfill-php70 (v1.10.0): Downloading (100%)
beam_tracker_1   | 2019/06/17 16:39:50 [EROR] unable to cache entity schemas err=Error 1146: Table 'beam.entities' doesn't exist
beam_tracker_1   | unable to cache entity schemas from MySQL
beam_tracker_1   | gitlab.com/remp/remp/Beam/go/model.(*EntitySchemaDB).Cache
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/model/entity_schema.go:86
beam_tracker_1   | main.main.func2
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/tracker/main.go:105
beam_tracker_1   | main.main.func4
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/tracker/main.go:133
beam_tracker_1   | runtime.goexit
beam_tracker_1   |      /usr/local/go/src/runtime/asm_amd64.s:1333
campaign_1       |   - Installing php-cs-fixer/diff (v1.3.0): Downloading (100%)
beam_segments_1  | 2019/06/17 16:39:54 [EROR] unable to cache segments err=Error 1146: Table 'beam.segments' doesn't exist
beam_segments_1  | unable to cache segments from MySQL
beam_segments_1  | gitlab.com/remp/remp/Beam/go/model.(*SegmentDB).Cache
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/model/segment.go:553
beam_segments_1  | main.main.func1
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/segments/main.go:112
beam_segments_1  | main.main.func5
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/segments/main.go:143
beam_segments_1  | runtime.goexit
beam_segments_1  |      /usr/local/go/src/runtime/asm_amd64.s:1333
beam_segments_1  | ELASTIC 2019/06/17 16:39:54 elastic: http://elasticsearch:9200 is dead
sso_1            |   - Installing symfony/options-resolver (v4.2.4): Downloading (100%)
campaign_1       |   - Installing doctrine/annotations (v1.6.0): Downloading (100%)
beam_tracker_1   | 2019/06/17 16:40:00 [EROR] unable to cache properties err=Error 1146: Table 'beam.properties' doesn't exist
beam_tracker_1   | unable to cache properties from MySQL
beam_tracker_1   | gitlab.com/remp/remp/Beam/go/model.(*PropertyDB).Cache
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/model/property.go:67
beam_tracker_1   | main.main.func1
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/tracker/main.go:97
beam_tracker_1   | main.main.func3
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/tracker/main.go:118
beam_tracker_1   | runtime.goexit
beam_tracker_1   |      /usr/local/go/src/runtime/asm_amd64.s:1333
sso_1            |   - Installing php-cs-fixer/diff (v1.3.0): Downloading (100%)
beam_segments_1  | 2019/06/17 16:40:04 [EROR] unable to cache segments err=Error 1146: Table 'beam.segments' doesn't exist
beam_segments_1  | unable to cache segments from MySQL
beam_segments_1  | gitlab.com/remp/remp/Beam/go/model.(*SegmentDB).Cache
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/model/segment.go:553
beam_segments_1  | main.main.func1
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/segments/main.go:112
beam_segments_1  | main.main.func5
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/segments/main.go:143
beam_segments_1  | runtime.goexit
beam_segments_1  |      /usr/local/go/src/runtime/asm_amd64.s:1333
beam_segments_1  | ELASTIC 2019/06/17 16:40:04 elastic: http://elasticsearch:9200 is dead
campaign_1       |   - Installing friendsofphp/php-cs-fixer (v2.14.0): Downloading (100%)
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:06Z","tags":["status","plugin:xpack_main@6.2.2","error"],"pid":1,"state":"red","message":"Status changed from green to red - Request Timeout after 3000ms","prevState":"
green","prevMsg":"Ready"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:06Z","tags":["status","plugin:searchprofiler@6.2.2","error"],"pid":1,"state":"red","message":"Status changed from green to red - Request Timeout after 3000ms","prevStat
e":"green","prevMsg":"Ready"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:06Z","tags":["status","plugin:ml@6.2.2","error"],"pid":1,"state":"red","message":"Status changed from green to red - Request Timeout after 3000ms","prevState":"green","
prevMsg":"Ready"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:06Z","tags":["status","plugin:tilemap@6.2.2","error"],"pid":1,"state":"red","message":"Status changed from green to red - Request Timeout after 3000ms","prevState":"gre
en","prevMsg":"Ready"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:07Z","tags":["status","plugin:watcher@6.2.2","error"],"pid":1,"state":"red","message":"Status changed from green to red - Request Timeout after 3000ms","prevState":"gre
en","prevMsg":"Ready"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:07Z","tags":["status","plugin:graph@6.2.2","error"],"pid":1,"state":"red","message":"Status changed from green to red - Request Timeout after 3000ms","prevState":"green
","prevMsg":"Ready"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:07Z","tags":["status","plugin:reporting@6.2.2","error"],"pid":1,"state":"red","message":"Status changed from green to red - Request Timeout after 3000ms","prevState":"g
reen","prevMsg":"Ready"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:07Z","tags":["status","plugin:security@6.2.2","error"],"pid":1,"state":"red","message":"Status changed from green to red - Request Timeout after 3000ms","prevState":"gr
een","prevMsg":"Ready"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:07Z","tags":["status","plugin:logstash@6.2.2","error"],"pid":1,"state":"red","message":"Status changed from green to red - Request Timeout after 3000ms","prevState":"gr
een","prevMsg":"Ready"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:07Z","tags":["info","monitoring-ui","kibana-monitoring"],"pid":1,"message":"Stopping all Kibana monitoring collectors"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:07Z","tags":["status","plugin:elasticsearch@6.2.2","error"],"pid":1,"state":"red","message":"Status changed from green to red - Request Timeout after 3000ms","prevState
":"green","prevMsg":"Ready"}
sso_1            |   - Installing doctrine/annotations (v1.6.0): Downloading (100%)
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:08Z","tags":["status","plugin:xpack_main@6.2.2","info"],"pid":1,"state":"green","message":"Status changed from red to green - Ready","prevState":"red","prevMsg":"Reques
t Timeout after 3000ms"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:08Z","tags":["status","plugin:searchprofiler@6.2.2","info"],"pid":1,"state":"green","message":"Status changed from red to green - Ready","prevState":"red","prevMsg":"Re
quest Timeout after 3000ms"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:09Z","tags":["status","plugin:ml@6.2.2","info"],"pid":1,"state":"green","message":"Status changed from red to green - Ready","prevState":"red","prevMsg":"Request Timeou
t after 3000ms"}
beam_tracker_1   | 2019/06/17 16:40:10 [EROR] unable to cache entity schemas err=Error 1146: Table 'beam.entities' doesn't exist
beam_tracker_1   | unable to cache entity schemas from MySQL
beam_tracker_1   | gitlab.com/remp/remp/Beam/go/model.(*EntitySchemaDB).Cache
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/model/entity_schema.go:86
beam_tracker_1   | main.main.func2
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/tracker/main.go:105
beam_tracker_1   | main.main.func4
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/tracker/main.go:133
beam_tracker_1   | runtime.goexit
beam_tracker_1   |      /usr/local/go/src/runtime/asm_amd64.s:1333
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:09Z","tags":["status","plugin:tilemap@6.2.2","info"],"pid":1,"state":"green","message":"Status changed from red to green - Ready","prevState":"red","prevMsg":"Request T
imeout after 3000ms"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:10Z","tags":["status","plugin:watcher@6.2.2","info"],"pid":1,"state":"green","message":"Status changed from red to green - Ready","prevState":"red","prevMsg":"Request T
imeout after 3000ms"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:10Z","tags":["status","plugin:graph@6.2.2","info"],"pid":1,"state":"green","message":"Status changed from red to green - Ready","prevState":"red","prevMsg":"Request Tim
eout after 3000ms"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:10Z","tags":["status","plugin:reporting@6.2.2","info"],"pid":1,"state":"green","message":"Status changed from red to green - Ready","prevState":"red","prevMsg":"Request
 Timeout after 3000ms"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:11Z","tags":["status","plugin:security@6.2.2","info"],"pid":1,"state":"green","message":"Status changed from red to green - Ready","prevState":"red","prevMsg":"Request
Timeout after 3000ms"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:11Z","tags":["status","plugin:logstash@6.2.2","info"],"pid":1,"state":"green","message":"Status changed from red to green - Ready","prevState":"red","prevMsg":"Request
Timeout after 3000ms"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:12Z","tags":["info","monitoring-ui","kibana-monitoring"],"pid":1,"message":"Starting all Kibana monitoring collectors"}
kibana_1         | {"type":"log","@timestamp":"2019-06-17T16:40:12Z","tags":["status","plugin:elasticsearch@6.2.2","info"],"pid":1,"state":"green","message":"Status changed from red to green - Ready","prevState":"red","prevMsg":"Req
uest Timeout after 3000ms"}
sso_1            |   - Installing friendsofphp/php-cs-fixer (v2.14.2): Downloading (100%)
beam_segments_1  | 2019/06/17 16:40:13 [EROR] unable to cache segments err=Error 1146: Table 'beam.segments' doesn't exist
beam_segments_1  | unable to cache segments from MySQL
beam_segments_1  | gitlab.com/remp/remp/Beam/go/model.(*SegmentDB).Cache
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/model/segment.go:553
beam_segments_1  | main.main.func1
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/segments/main.go:112
beam_segments_1  | main.main.func5
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/segments/main.go:143
beam_segments_1  | runtime.goexit
beam_segments_1  |      /usr/local/go/src/runtime/asm_amd64.s:1333
beam_tracker_1   | 2019/06/17 16:40:20 [EROR] unable to cache properties err=Error 1146: Table 'beam.properties' doesn't exist
beam_tracker_1   | unable to cache properties from MySQL
beam_tracker_1   | gitlab.com/remp/remp/Beam/go/model.(*PropertyDB).Cache
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/model/property.go:67
beam_tracker_1   | main.main.func1
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/tracker/main.go:97
beam_tracker_1   | main.main.func3
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/tracker/main.go:118
beam_tracker_1   | runtime.goexit
beam_tracker_1   |      /usr/local/go/src/runtime/asm_amd64.s:1333
campaign_1       |   - Installing hamcrest/hamcrest-php (v2.0.0): Downloading (100%)
elasticsearch_1  | [2019-06-17T16:40:24,038][INFO ][o.e.m.j.JvmGcMonitorService] [nEUG9XY] [gc][295] overhead, spent [289ms] collecting in the last [1s]
beam_segments_1  | 2019/06/17 16:40:24 [EROR] unable to cache segments err=Error 1146: Table 'beam.segments' doesn't exist
beam_segments_1  | unable to cache segments from MySQL
beam_segments_1  | gitlab.com/remp/remp/Beam/go/model.(*SegmentDB).Cache
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/model/segment.go:553
beam_segments_1  | main.main.func1
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/segments/main.go:112
beam_segments_1  | main.main.func5
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/segments/main.go:143
beam_segments_1  | runtime.goexit
beam_segments_1  |      /usr/local/go/src/runtime/asm_amd64.s:1333
beam_segments_1  | ELASTIC 2019/06/17 16:40:34 elastic: http://elasticsearch:9200 is dead
beam_segments_1  | 2019/06/17 16:40:35 [EROR] unable to cache segments err=Error 1146: Table 'beam.segments' doesn't exist
beam_segments_1  | unable to cache segments from MySQL
beam_segments_1  | gitlab.com/remp/remp/Beam/go/model.(*SegmentDB).Cache
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/model/segment.go:553
beam_segments_1  | main.main.func1
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/segments/main.go:112
beam_segments_1  | main.main.func5
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/segments/main.go:143
beam_segments_1  | runtime.goexit
beam_segments_1  |      /usr/local/go/src/runtime/asm_amd64.s:1333
beam_tracker_1   | 2019/06/17 16:40:31 [EROR] unable to cache entity schemas err=Error 1146: Table 'beam.entities' doesn't exist
elasticsearch_1  | [2019-06-17T16:40:38,283][INFO ][o.e.m.j.JvmGcMonitorService] [nEUG9XY] [gc][young][302][38] duration [963ms], collections [1]/[1.6s], total [963ms]/[23.6s], memory [245.3mb]->[173.3mb]/[1007.3mb], all_pools {[you
ng] [79.5mb]->[15.4mb]/[133.1mb]}{[survivor] [11.9mb]->[16.6mb]/[16.6mb]}{[old] [153.8mb]->[156.2mb]/[857.6mb]}
elasticsearch_1  | [2019-06-17T16:40:38,283][WARN ][o.e.m.j.JvmGcMonitorService] [nEUG9XY] [gc][302] overhead, spent [963ms] collecting in the last [1.6s]
beam_tracker_1   | unable to cache entity schemas from MySQL
beam_tracker_1   | gitlab.com/remp/remp/Beam/go/model.(*EntitySchemaDB).Cache
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/model/entity_schema.go:86
mailhog_1        | [APIv1] KEEPALIVE /api/v1/events
beam_tracker_1   | main.main.func2
elasticsearch_1  | [2019-06-17T16:40:41,624][INFO ][o.e.m.j.JvmGcMonitorService] [nEUG9XY] [gc][young][304][39] duration [878ms], collections [1]/[2.1s], total [878ms]/[24.5s], memory [288.3mb]->[169.1mb]/[1007.3mb], all_pools {[you
ng] [115.4mb]->[2.9mb]/[133.1mb]}{[survivor] [16.6mb]->[11.9mb]/[16.6mb]}{[old] [156.2mb]->[157.1mb]/[857.6mb]}
elasticsearch_1  | [2019-06-17T16:40:41,624][INFO ][o.e.m.j.JvmGcMonitorService] [nEUG9XY] [gc][304] overhead, spent [878ms] collecting in the last [2.1s]
sso_1            |   - Installing hamcrest/hamcrest-php (v2.0.0): Downloading (100%)
campaign_1       |   - Installing mockery/mockery (1.2.2): Downloading (100%)
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/tracker/main.go:105
beam_tracker_1   | main.main.func4
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/tracker/main.go:133
beam_tracker_1   | runtime.goexit
beam_tracker_1   |      /usr/local/go/src/runtime/asm_amd64.s:1333
beam_tracker_1   | 2019/06/17 16:40:41 [EROR] unable to cache properties err=Error 1146: Table 'beam.properties' doesn't exist
beam_tracker_1   | unable to cache properties from MySQL
beam_tracker_1   | gitlab.com/remp/remp/Beam/go/model.(*PropertyDB).Cache
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/model/property.go:67
beam_tracker_1   | main.main.func1
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/tracker/main.go:97
beam_tracker_1   | main.main.func3
beam_tracker_1   |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/tracker/main.go:118
beam_tracker_1   | runtime.goexit
beam_tracker_1   |      /usr/local/go/src/runtime/asm_amd64.s:1333
beam_segments_1  | ELASTIC 2019/06/17 16:40:44 elastic: http://elasticsearch:9200 is dead
beam_segments_1  | 2019/06/17 16:40:44 [EROR] unable to cache segments err=Error 1146: Table 'beam.segments' doesn't exist
beam_segments_1  | unable to cache segments from MySQL
beam_segments_1  | gitlab.com/remp/remp/Beam/go/model.(*SegmentDB).Cache
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/model/segment.go:553
beam_segments_1  | main.main.func1
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/segments/main.go:112
beam_segments_1  | main.main.func5
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/segments/main.go:143
beam_segments_1  | runtime.goexit
beam_segments_1  |      /usr/local/go/src/runtime/asm_amd64.s:1333
elasticsearch_1  | [2019-06-17T16:40:56,795][WARN ][o.e.m.j.JvmGcMonitorService] [nEUG9XY] [gc][young][310][40] duration [1.7s], collections [1]/[6.5s], total [1.7s]/[26.3s], memory [280.7mb]->[166.5mb]/[1007.3mb], all_pools {[young
] [111.7mb]->[4.2mb]/[133.1mb]}{[survivor] [11.9mb]->[9.4mb]/[16.6mb]}{[old] [157.1mb]->[157.1mb]/[857.6mb]}
beam_segments_1  | ELASTIC 2019/06/17 16:40:56 elastic: http://elasticsearch:9200 is dead
elasticsearch_1  | [2019-06-17T16:40:57,376][INFO ][o.e.m.j.JvmGcMonitorService] [nEUG9XY] [gc][310] overhead, spent [1.7s] collecting in the last [6.5s]
beam_tracker_1   | 2019/06/17 16:41:02 [EROR] unable to cache entity schemas err=Error 1146: Table 'beam.entities' doesn't exist
beam_tracker_1   | unable to cache entity schemas from MySQL
beam_segments_1  | ELASTIC 2019/06/17 16:41:07 elastic: http://elasticsearch:9200 is dead
beam_segments_1  | 2019/06/17 16:41:08 [EROR] unable to cache segments err=Error 1146: Table 'beam.segments' doesn't exist
beam_segments_1  | unable to cache segments from MySQL
beam_segments_1  | gitlab.com/remp/remp/Beam/go/model.(*SegmentDB).Cache
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/model/segment.go:553
beam_segments_1  | main.main.func1
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/segments/main.go:112
beam_segments_1  | main.main.func5
beam_segments_1  |      /go/src/gitlab.com/remp/remp/Beam/go/cmd/segments/main.go:143
beam_segments_1  | runtime.goexit
beam_segments_1  |      /usr/local/go/src/runtime/asm_amd64.s:1333
beam_tracker_1   | gitlab.com/remp/remp/Beam/go/model.(*EntitySchemaDB).Cache
```

**Comment by PayteR** [comment]
Hmm i think it's not possible because there is not possible to run latest Docker Daemon on WSL https://github.com/microsoft/WSL/issues/2291 one that works is older version  Docker-CE 17.09.0 and older Ubuntu for WSL 16.0.4 LTS https://medium.com/faun/docker-running-seamlessly-in-windows-subsystem-linux-6ef8412377aa i managed that, but there is another problem with a newer version of the docker-compose.yml file 

```
ERROR: Version in "./docker-compose.yml" is unsupported. You might be seeing this error because you're using the wrong Compose file version. Either specify a version of "2" (or "2.0") and place your service definitions under the `services` key, or omit the `version` key and place your service definitions at the root of the file to use version 1.
For more on the Compose file format versions, see https://docs.docker.com/compose/compose-file/
```
When i change file version to 2 then this error happens
```
ERROR: Couldn't connect to Docker daemon at http+docker://localunixsocket - is it running?

If it's at a non-standard location, specify the URL with the DOCKER_HOST environment variable.
```
So I'm really stuck here, what's the solution? Thx

**Comment by rootpd** [comment]
Hi,

the errors you were seing were "fine" during installation period. Go APIs are trying to connect to DB which wasn't initialized yet. In the background, there was a process that installed composer/yarn dependencies for all 4 of included web apps. You can see references in the log too:

```
sso_1            |   - Installing hamcrest/hamcrest-php (v2.0.0): Downloading (100%)
campaign_1       |   - Installing mockery/mockery (1.2.2): Downloading (100%)
```

What I recommend now is to run `docker-compose down`, checkout the repository again and run the clean install. The checkout is important because installation is run only once - when `.env` files do not exist yet. When you run the `docker-compose up -d` for the first time, let it do its thing. 

You can follow the installation process by checking logs of specific web apps. Once they're not generating anything, the installation should be ready:

```
docker-compose logs beam -f
docker-compose logs campaign -f
docker-compose logs sso -f
docker-compose logs mailer -f
```

The installation is running [remp.sh](https://github.com/remp2020/remp/blob/master/Docker/php/remp.sh) installation script - just so you have an idea of the amount of things it's doing.

**Comment by rootpd** [comment]
As for the WSL, please skip that entirely. As you wrote, it's not possible to run Docker in WSL yet. There is a way how to connect WSL docker commands to Windows Docker installation but I wouldn't recommend it.

Just run the manual installation commands I mentioned earlier to generate `.tar` files and then run `docker-compose up -d` within Windows shell.

**Comment by PayteR** [comment]
I cleared all by `docker-compose down -v`, then i removed whole repository and cloned all again. Then i deleted my Ubuntu WSL installation and installed docker there for `make docker-build` command. But i ended up with this error:
```
make docker-build
cd "Beam" && make docker-build
make[1]: Entering directory '/mnt/d/projects/webnoviny/wbn_crm/Beam'
make[2]: Entering directory '/mnt/d/projects/webnoviny/wbn_crm/Beam/go/cmd/tracker'
set -e; \
docker build -t remp-beam_tracker_builder builder; \
docker run -e TARGET_GOOS -e TARGET_GOARCH -e TARGET_SUFFIX --rm -v $PWD/../../:/src/build remp-beam_tracker_builder > tracker.tar;
Sending build context to Docker daemon  3.072kB
Step 1/8 : FROM golang:1.11
1.11: Pulling from library/golang
65014b3c3121: Pull complete
266cffe9d908: Pull complete
2c0c4ec7a15b: Pull complete
7d8084e6e79c: Pull complete
2258ad2576bb: Pull complete
c860e60ff524: Pull complete
3a78ccd9d205: Pull complete
17bb5b24ea72: Pull complete
0d5c437aa873: Pull complete
dad53d888c2a: Pull complete
7c12abd45e5f: Pull complete
49b305932247: Pull complete
942f854409c1: Pull complete
Digest: sha256:6d762eb2c24807a5e38fe231e1836b4c96f5961d4dbdaea5a19267567a5e6834
Status: Downloaded newer image for golang:1.11
 ---> 99e6225e2cf0
Step 2/8 : RUN go get golang.org/x/tools/cmd/goimports
 ---> Running in f7f211ccf095
Removing intermediate container f7f211ccf095
 ---> a2f85806c605
Step 3/8 : RUN mkdir -p /src/build
 ---> Running in dadb8e50d947


    Directory: C:\src


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----        6/26/2019  10:16 PM                build


Removing intermediate container dadb8e50d947
 ---> 517fd23d23f3
Step 4/8 : RUN mkdir -p /go/src/gitlab.com/remp/remp/Beam/go
 ---> Running in a4aad3769b6a


    Directory: C:\go\src\gitlab.com\remp\remp\Beam


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----        6/26/2019  10:16 PM                go


Removing intermediate container a4aad3769b6a
 ---> 4afc13fd8b12
Step 5/8 : WORKDIR /go/src/gitlab.com/remp/remp/Beam/go
Removing intermediate container fc7483b9e137
 ---> d45086d23286
Step 6/8 : COPY build /usr/local/bin/
 ---> dfc2e387ba65
Step 7/8 : RUN chmod +x /usr/local/bin/build
 ---> Running in 2332e02a9766
chmod : The term 'chmod' is not recognized as the name of a cmdlet, function,
script file, or operable program. Check the spelling of the name, or if a path
was included, verify that the path is correct and try again.
At line:1 char:76
+ ... nce = 'Stop'; $ProgressPreference = 'SilentlyContinue'; chmod +x /usr ...
+                                                             ~~~~~
    + CategoryInfo          : ObjectNotFound: (chmod:String) [], ParentContain
   sErrorRecordException
    + FullyQualifiedErrorId : CommandNotFoundException

The command 'powershell -Command $ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue'; chmod +x /usr/local/bin/build' returned a non-zero code: 1
Makefile:40: recipe for target 'docker-build' failed
make[2]: *** [docker-build] Error 1
make[2]: Leaving directory '/mnt/d/projects/webnoviny/wbn_crm/Beam/go/cmd/tracker'
Makefile:42: recipe for target 'docker-build' failed
make[1]: *** [docker-build] Error 2
make[1]: Leaving directory '/mnt/d/projects/webnoviny/wbn_crm/Beam'
Makefile:41: recipe for target 'docker-build' failed
make: *** [docker-build] Error 2
```
I didn't have a problem with make command before, did you changed something in recipies lately?

**Comment by PayteR** [comment]
I cleared all by `docker-compose down -v`, then i removed whole repository and cloned all again. Then i deleted my Ubuntu WSL installation and installed docker there for `make docker-build` command. But i ended up with this error:
```
make docker-build
cd "Beam" && make docker-build
make[1]: Entering directory '/mnt/d/projects/webnoviny/wbn_crm/Beam'
make[2]: Entering directory '/mnt/d/projects/webnoviny/wbn_crm/Beam/go/cmd/tracker'
set -e; \
docker build -t remp-beam_tracker_builder builder; \
docker run -e TARGET_GOOS -e TARGET_GOARCH -e TARGET_SUFFIX --rm -v $PWD/../../:/src/build remp-beam_tracker_builder > tracker.tar;
Sending build context to Docker daemon  3.072kB
Step 1/8 : FROM golang:1.11
1.11: Pulling from library/golang
65014b3c3121: Pull complete
266cffe9d908: Pull complete
2c0c4ec7a15b: Pull complete
7d8084e6e79c: Pull complete
2258ad2576bb: Pull complete
c860e60ff524: Pull complete
3a78ccd9d205: Pull complete
17bb5b24ea72: Pull complete
0d5c437aa873: Pull complete
dad53d888c2a: Pull complete
7c12abd45e5f: Pull complete
49b305932247: Pull complete
942f854409c1: Pull complete
Digest: sha256:6d762eb2c24807a5e38fe231e1836b4c96f5961d4dbdaea5a19267567a5e6834
Status: Downloaded newer image for golang:1.11
 ---> 99e6225e2cf0
Step 2/8 : RUN go get golang.org/x/tools/cmd/goimports
 ---> Running in f7f211ccf095
Removing intermediate container f7f211ccf095
 ---> a2f85806c605
Step 3/8 : RUN mkdir -p /src/build
 ---> Running in dadb8e50d947


    Directory: C:\src


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----        6/26/2019  10:16 PM                build


Removing intermediate container dadb8e50d947
 ---> 517fd23d23f3
Step 4/8 : RUN mkdir -p /go/src/gitlab.com/remp/remp/Beam/go
 ---> Running in a4aad3769b6a


    Directory: C:\go\src\gitlab.com\remp\remp\Beam


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----        6/26/2019  10:16 PM                go


Removing intermediate container a4aad3769b6a
 ---> 4afc13fd8b12
Step 5/8 : WORKDIR /go/src/gitlab.com/remp/remp/Beam/go
Removing intermediate container fc7483b9e137
 ---> d45086d23286
Step 6/8 : COPY build /usr/local/bin/
 ---> dfc2e387ba65
Step 7/8 : RUN chmod +x /usr/local/bin/build
 ---> Running in 2332e02a9766
chmod : The term 'chmod' is not recognized as the name of a cmdlet, function,
script file, or operable program. Check the spelling of the name, or if a path
was included, verify that the path is correct and try again.
At line:1 char:76
+ ... nce = 'Stop'; $ProgressPreference = 'SilentlyContinue'; chmod +x /usr ...
+                                                             ~~~~~
    + CategoryInfo          : ObjectNotFound: (chmod:String) [], ParentContain
   sErrorRecordException
    + FullyQualifiedErrorId : CommandNotFoundException

The command 'powershell -Command $ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue'; chmod +x /usr/local/bin/build' returned a non-zero code: 1
Makefile:40: recipe for target 'docker-build' failed
make[2]: *** [docker-build] Error 1
make[2]: Leaving directory '/mnt/d/projects/webnoviny/wbn_crm/Beam/go/cmd/tracker'
Makefile:42: recipe for target 'docker-build' failed
make[1]: *** [docker-build] Error 2
make[1]: Leaving directory '/mnt/d/projects/webnoviny/wbn_crm/Beam'
Makefile:41: recipe for target 'docker-build' failed
make: *** [docker-build] Error 2
```
I didn't have a problem with make command before, did you changed something in recipies lately?

**Comment by PayteR** [comment]
After i restared PC i have run make command again, and now there is this error
```
make docker-build
cd "Beam" && make docker-build
make[1]: Entering directory '/mnt/d/projects/webnoviny/wbn_crm/Beam'
make[2]: Entering directory '/mnt/d/projects/webnoviny/wbn_crm/Beam/go/cmd/tracker'
set -e; \
docker build -t remp-beam_tracker_builder builder; \
docker run -e TARGET_GOOS -e TARGET_GOARCH -e TARGET_SUFFIX --rm -v $PWD/../../:/src/build remp-beam_tracker_builder > tracker.tar;
Sending build context to Docker daemon  3.072kB
Step 1/8 : FROM golang:1.11
1.11: Pulling from library/golang
6f2f362378c5: Pull complete
494c27a8a6b8: Pull complete
7596bb83081b: Pull complete
372744b62d49: Pull complete
0658c6765517: Pull complete
f21bd173b2c4: Pull complete
4469a0e66964: Pull complete
Digest: sha256:6d762eb2c24807a5e38fe231e1836b4c96f5961d4dbdaea5a19267567a5e6834
Status: Downloaded newer image for golang:1.11
 ---> 5b555dd4804a
Step 2/8 : RUN go get golang.org/x/tools/cmd/goimports
 ---> Running in d802cf9f0612
Removing intermediate container d802cf9f0612
 ---> c62853d49d3e
Step 3/8 : RUN mkdir -p /src/build
 ---> Running in 1d942ae2c028
Removing intermediate container 1d942ae2c028
 ---> a6bd3b1ef24c
Step 4/8 : RUN mkdir -p /go/src/gitlab.com/remp/remp/Beam/go
 ---> Running in 429618610b1f
Removing intermediate container 429618610b1f
 ---> f1009b10fff7
Step 5/8 : WORKDIR /go/src/gitlab.com/remp/remp/Beam/go
Removing intermediate container ced44b45aafb
 ---> efbef65b44cb
Step 6/8 : COPY build /usr/local/bin/
 ---> 1453b317c96a
Step 7/8 : RUN chmod +x /usr/local/bin/build
 ---> Running in 832076dc98dc
Removing intermediate container 832076dc98dc
 ---> cae5c230f526
Step 8/8 : CMD ["build"]
 ---> Running in 2f6fd4bcd3c2
Removing intermediate container 2f6fd4bcd3c2
 ---> 28911b36635b
Successfully built 28911b36635b
Successfully tagged remp-beam_tracker_builder:latest
/usr/local/bin/build: line 5: cd: cmd/tracker: No such file or directory
tar: tracker: Cannot stat: No such file or directory
tar: swagger: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
Makefile:40: recipe for target 'docker-build' failed
make[2]: *** [docker-build] Error 2
make[2]: Leaving directory '/mnt/d/projects/webnoviny/wbn_crm/Beam/go/cmd/tracker'
Makefile:42: recipe for target 'docker-build' failed
make[1]: *** [docker-build] Error 2
make[1]: Leaving directory '/mnt/d/projects/webnoviny/wbn_crm/Beam'
Makefile:41: recipe for target 'docker-build' failed
make: *** [docker-build] Error 2
```
